v0/destinations/zendesk/transform.js

```diff
@@ -240,6 +227,48 @@ function getIdentifyPayload(message, category, destinationConfig, type) {
 
   return payload;
 }
+/**
+ * ref: https://developer.zendesk.com/api-reference/ticketing/users/users/#search-users
+ * @param {*} message message
+ * @param {*} headers headers for authorizations
+ * @returns
+ */
+const getUserIdByExternalId = async (message, headers) => {
+  const traits = getFieldValueFromMessage(message, "traits");
```

Comment:
We are already getting traits from where we are calling `getUserIdByExternalId` function. You can pass the traits as a parameter from there.

Reply:
updated